cli/src/formatter.rs

```diff
@@ -473,19 +480,43 @@ impl<W: Write> Write for ColorFormatter<W> {
          * Some tools (like `less -R`) get confused and lose coloring of lines
            after a newline.
          */
+
         for line in data.split_inclusive(|b| *b == b'\n') {
             if line.ends_with(b"\n") {
                 self.write_new_style()?;
-                write_sanitized(&mut self.output, &line[..line.len() - 1])?;
+                write_line_exclusive(
+                    &mut self.output,
+                    &line[..line.len() - 1],
+                    &self.labels,
+                    self.debug,
+                )?;
                 let labels = mem::take(&mut self.labels);
                 self.write_new_style()?;
                 self.output.write_all(b"\n")?;
                 self.labels = labels;
             } else {
                 self.write_new_style()?;
-                write_sanitized(&mut self.output, line)?;
+                write_line_exclusive(&mut self.output, line, &self.labels, self.debug)?;
             }
         }
+
+        fn write_line_exclusive<W: Write>(
+            output: &mut W,
+            buf: &[u8],
+            labels: &[String],
+            debug: bool,
+        ) -> io::Result<()> {
```

Comment:
(at your option) this can be implemented as `self.write_line_exclusive(buf)`.

Reply:
I figure it's only used there for now and need not be accessible elsewhere. If the need arises, someone can refactor this.